docs/update_by_query.rst

```diff
@@ -0,0 +1,148 @@
+.. _update_by_query:
+
+Update By Query
+================
+
+The ``Update By Query`` object
+-------------------------------
+
+The ``Update By Query`` object enables the use of the 
+`_update_by_query <https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html>`_
+endpoint to perform an update on documents that match a search query.
+
+The object is implemented as a modification of the ``Search`` object, containing a 
+subset of its query methods, as well as a script method, which is used to make updates.
+
+The ``Update By Query`` object implements the following ``Search`` query types:
+  
+  * queries
+
+  * filters
+
+  * excludes
+
+For more information on queries, see the :ref:`search_dsl` chapter.
+
+Like the ``Search`` object, the API is designed to be chainable. This means that the ``Update By Query`` object
+is immutable: all changes to the object will result in a shallow copy being created which
+contains the changes. This means you can safely pass the ``Update By Query`` object to
+foreign code without fear of it modifying your objects as long as it sticks to
+the ``Update By Query`` object APIs.
+
+You can pass an instance of the low-level `elasticsearch client <https://elasticsearch-py.readthedocs.io/>`_ when
+instantiating the ``Update By Query`` object:
```

Comment:
can we instead point to the `configuration` chapter to promote global configuration instead of passing the client everywhere explicitly?

i know you mention it lower but that should be the recommended way and should be more highlighted.